gdal/ogr/ogrfielddefn.cpp

```diff
@@ -224,6 +227,95 @@ const char *OGR_Fld_GetNameRef( OGRFieldDefnH hDefn )
     return OGRFieldDefn::FromHandle(hDefn)->GetNameRef();
 }
 
+
+/************************************************************************/
+/*                              SetAlias()                               */
+/************************************************************************/
+
+/**
+ * \brief Reset the alias for this field.
+ *
+ * This method is the same as the C function OGR_Fld_SetAlias().
+ *
+ * @param pszAliasIn the new alias to apply.
+ *
+ * @since GDAL 3.2
+ */
+
+void OGRFieldDefn::SetAlias( const char * pszAliasIn )
+
+{
+    if( pszAlias != pszAliasIn )
+    {
+        CPLFree(pszAlias);
+        pszAlias = CPLStrdup(pszAliasIn);
+    }
+}
+
+/************************************************************************/
+/*                          OGR_Fld_SetAlias()                           */
+/************************************************************************/
+/**
+ * \brief Reset the alias for this field.
+ *
+ * This function is the same as the CPP method OGRFieldDefn::SetAlias().
+ *
+ * @param hDefn handle to the field definition to apply the new alias to.
+ * @param pszAlias the new alias to apply.
+ *
+ * @since GDAL 3.2
+ */
+
+void OGR_Fld_SetAlias( OGRFieldDefnH hDefn, const char *pszAlias )
+
+{
+    OGRFieldDefn::FromHandle(hDefn)->SetAlias(pszAlias);
+}
+
+/************************************************************************/
+/*                             GetAliasRef()                             */
+/************************************************************************/
+
+/**
+ * \fn const char *OGRFieldDefn::GetAliasRef();
+ *
+ * \brief Fetch alias for this field.
```

Comment:
We should probably add more about the semantics of what "alias". See above point.